src/v/config/property.h

```diff
@@ -953,5 +1053,122 @@ class hidden_when_default_property : public property<T> {
         return this->value() == this->default_value();
     }
 };
+namespace detail {
+
+template<typename P>
+concept Property = requires() {
+    std::derived_from<P, base_property>;
+    typename P::underlying;
+};
+
+template<typename T>
+concept Array = !(std::is_same_v<std::decay_t<T>, ss::sstring>
+                  || std::is_same_v<std::decay_t<T>, std::string>)
+                && is_collection<T>;
+
+} // namespace detail
+
+template<detail::Property P>
+class enterprise : public P {
+    template<typename T, typename = void>
```

Comment:
where can I learn more about this? my gut feeling is that having enterprise policies represented in the configuration system won't turn out well (with some exceptions, like simple visibility policies--but then i'm surprised to see "enterprise" word at all).

Reply:
This is the PR for it #23899
And this is the corresponding ticket [CORE-7997](https://redpandadata.atlassian.net/browse/CORE-7997)

Reply:
I've been asking the docs team to request core team to break out configuration metadata into something like YAML so that it is easier to change things like doc strings, and read with other tools. Would the introduction of "enterprise properties" make this difficult (e.g. by being hard to represent)?

Reply:
> I've been asking the docs team to request core team to break out configuration metadata into something like YAML so that it is easier to change things like doc strings, and read with other tools. Would the introduction of "enterprise properties" make this difficult (e.g. by being hard to represent)?

It's a good question!  Can we move it to #23899?

Reply:
yeh i dunno what goes where. tag me

Reply:
I added you to the review in #23899 which defines this `enterprise` property type

Reply:
thanks!